src/main/java/jenkins/plugins/slack/webhook/WebhookHandler.java

```diff
@@ -0,0 +1,139 @@
+            SecurityContextHolder.setContext(ctx);
+            return new SlackTextMessage("Could not find project ("+projectName+")\n");
+        }
+
+        AbstractBuild build =
+            project.getBuildByNumber(Integer.parseInt(buildNumber));
+
+        if (build == null) {
+            SecurityContextHolder.setContext(ctx);
+            return new SlackTextMessage("Could not find build #"+buildNumber+" for ("+projectName+")\n");
+        }
+
+        SecurityContextHolder.setContext(ctx);
+
+        List<String> log = new ArrayList<String>();
+        try {
+            log = build.getLog(25);
+        } catch (IOException ex) {
+            return new SlackTextMessage("Error occured returning log: "+ex.getMessage());
+        }
+
+        String response = "*"+projectName+"* *#"+buildNumber+"*\n";
+        response += "```";
+        for (String line : log) {
+            response += line + "\n";
+        }
+        response += "```";
+
+        return new SlackTextMessage(response);
+    }
+
+    public SlackTextMessage scheduleJob(String projectName) {
+
+        SecurityContext ctx = ACL.impersonate(ACL.SYSTEM);
+
+        String response = "";
+
+        Project project =
+            Jenkins.getInstance().getItemByFullName(projectName, Project.class);
+
+        if (project == null) {
+            SecurityContextHolder.setContext(ctx);
+            return new SlackTextMessage("Could not find project ("+projectName+")\n");
+        }
+
+        if (project.scheduleBuild(new SlackWebhookCause(this.data.getUser_name()))) {
+            SecurityContextHolder.setContext(ctx);
+            return new SlackTextMessage("Build scheduled for project "+ projectName+"\n");
+        } else {
+            return new SlackTextMessage("Build not scheduled due to an issue with Jenkins");
+        }
+    }
+
+    public SlackTextMessage listProjects() {
+
+        SecurityContext ctx = ACL.impersonate(ACL.SYSTEM);
+
+        String response = "*Projects:*\n";
+
+        List<AbstractProject> jobs =
+            Jenkins.getInstance().getAllItems(AbstractProject.class);
+
+        SecurityContextHolder.setContext(ctx);
+
+        for (AbstractProject job : jobs) {
+            if (job.isBuildable()) {
+                AbstractBuild lastBuild = job.getLastBuild();
+                String buildNumber = "TBD";
+                String status = "TBD";
+                if (lastBuild != null) {
+
+                    buildNumber = Integer.toString(lastBuild.getNumber());
+
+                    if (lastBuild.isBuilding()) {
+                        status = "BUILDING";
+                    }
+
+                    Result result = lastBuild.getResult();
+
+                    if (result != null) {
+                        status = result.toString();
+                    }
+                }
+                response += ">*"+job.getDisplayName() + "*\n>*Last Build:* #"+buildNumber+"\n>*Status:* "+status;
+                response += "\n\n\n";
```

Comment:
Is there any concern about what this will look like when there are a lot of projects? This is quite a verbose format, which is nice for a few projects, but could result in a very long message.